src/providers/WalletProvider/WalletProvider.ts

```diff
@@ -1,14 +1,73 @@
-import * as React from 'react'
-import { DefaultProps, Props } from './WalletProvider.types'
+import React from 'react'
+import { Eth } from 'web3x-es/eth'
+import { EthereumProvider } from 'web3x-es/providers/ethereum-provider'
+import { Props } from './WalletProvider.types'
 
 export default class WalletProvider extends React.PureComponent<Props> {
-  static defaultProps: DefaultProps = {
-    children: null
+  eth = Eth.fromCurrentProvider()
+
+  // handle account change
+  handleChangeAccount = async () => {
+    if (!this.eth) return
+    const { onChangeAccount } = this.props
+    const accounts = await this.eth.getAccounts()
+    if (accounts.length > 0) {
+      const address = accounts[0].toString()
+      onChangeAccount(address)
+    }
+  }
+
+  // handle network change
+  handleChangeNetwork = async () => {
+    if (!this.eth) return
+    const { onChangeNetwork } = this.props
+    const network = await this.eth.getId()
+    onChangeNetwork(network)
+  }
+
+  handle(
+    action: 'on' | 'removeListener',
+    type: 'accountsChanged' | 'networkChanged',
+    handler: Function
+  ) {
+    // try to use web3x abstraction
+    if (this.eth) {
+      try {
+        this.eth.provider[action](type as any, handler as any)
```

Comment:
It is showing an error if we type `type` and `handler`, right?

Reply:
yes, because of the dynamic accessor `action` it can't be properly typed.